skytemple_files/common/ppmdu_config/pmdsky_debug/loader.py

```diff
@@ -0,0 +1,302 @@
+        return newobj
+
+
+def _read_symbol(loadaddr: int, symbol_def: dict, typ: Pmd2BinarySymbolType, region: str) -> Optional[Pmd2BinarySymbol]:
+    assert isinstance(loadaddr, int)
+    if 'name' in symbol_def:
+        name = symbol_def['name']
+    else:
+        raise ValueError("Symbol is missing its name.")
+    
+    if 'address' in symbol_def:
+        if region in symbol_def['address']:
+            if isinstance(symbol_def['address'][region], list):
+                begin = symbol_def['address'][region][0] - loadaddr
+            else:
+                begin = symbol_def['address'][region] - loadaddr
+        else:
+            return None
+    else:
+        raise ValueError(f"Symbol {name} is missing an address.")
+
+    end = None
+    if 'length' in symbol_def:
+        if region in symbol_def['length']:
+            end = begin + symbol_def['length'][region]
+    elif typ == Pmd2BinarySymbolType.DATA:
+        # We assume it has length 4 (pointer length) if not defined.
+        end = begin + 4
+
+    description = symbol_def.get('description', "")
+
+    return Pmd2BinarySymbol(
+        name=name,
+        typ=typ,
+        begin=begin,
+        end=end,
+        description=description
+    )
+
+
+def _read(binaries: Dict[str, _IncompleteBinary], yml: dict, region: str):
+    # arm9.bin and the overlays MUST be read first.
+    for bin_name, definition in yml.items():
+        assert isinstance(bin_name, str)
+        binary = _IncompleteBinary.get(binaries, bin_name)
+        if 'address' in definition:
+            if region in definition['address']:
+                if binary.loadaddress is None:
+                    binary.loadaddress = int(definition['address'][region])
+        if 'length' in definition:
+            if region in definition['length']:
+                if binary.length is None:
+                    binary.length = int(definition['length'][region])
+        if 'functions' in definition:
+            assert binary.loadaddress is not None
+            for symbol_def in definition['functions']:
+                sym = _read_symbol(binary.loadaddress, symbol_def, Pmd2BinarySymbolType.FUNCTION, region)
+                if sym is not None:
+                    binary.symbols.append(sym)
+        if 'data' in definition:
+            assert binary.loadaddress is not None
+            for symbol_def in definition['data']:
+                sym = _read_symbol(binary.loadaddress, symbol_def, Pmd2BinarySymbolType.DATA, region)
+                if sym is not None:
+                    binary.symbols.append(sym)
+        if 'description' in definition:
+            if binary.description == "":
+                binary.description = definition['description']
+
+
+def _build(binaries: Dict[str, _IncompleteBinary]) -> List[Pmd2Binary]:
+    out = []
+    for bin_name, incl_bin in binaries.items():
+        if incl_bin.loadaddress is None:
+            raise ValueError(f"Did not find a load address for {bin_name}.")
```

Comment:
Err, this isn't going to be called with `EoS_JP` is it? I don't actually know what the JP load addresses/binary sizes are so they aren't in `pmdsky-debug`.

Reply:
Not at the moment.